File: Avalon/WzData/MapData.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace WzDumper.WzData
{
    public struct Frame
    {
        public string bitmapPath;
        public int durationMs;
        public Point position;
    }

    public struct Tile
    {
        public List<Frame> frames;
    }

    public struct Layer
    {
        public List<Tile> tiles;
    }

    public struct Assets
    {
        public Dictionary<string, Bitmap> bitmaps; // BitmapPath -> Bitmap
    }

    public struct MapData
    {
        public Size mapSize;
        public List<Layer> layers;
    }

    public struct AvailableMap
    {
        public int mapId;
        public string mapStreetName;
        public string mapName;
    }

    public struct AvailableMaps
    {
        public List<AvailableMap> maps;
    }
}

File: Avalon/WzDumperLib/WzDumperLib/MapData.cpp
#include "MapDataConverters.hpp"
#include "TypesConverters.hpp"
#include "Converters.hpp"

WzDumperLib::Frame WzDumperLib::to_native(::WzDumper::Frame ^ managed)
{
    return {
        to_native(managed->bitmapPath),
        to_native(managed->durationMs),
        to_native(managed->position)
    };
}

WzDumperLib::Tile WzDumperLib::to_native(::WzDumper::Tile ^ managed)
{
    return {
        to_native_list(managed->frames)
    };
}

WzDumperLib::Layer WzDumperLib::to_native(::WzDumper::Layer ^ managed)
{
    return {
        to_native_list(managed->tiles)
    };
}

WzDumperLib::MapData WzDumperLib::to_native(::WzDumper::MapData ^ managed)
{
    return {
        to_native(managed->mapSize),
        to_native_list(managed->layers)
    };
}

WzDumperLib::AvailableMap WzDumperLib::to_native(::WzDumper::AvailableMap ^ managed)
{
    return {
        to_native(managed->mapId),
        to_native(managed->mapName)
    };
}
